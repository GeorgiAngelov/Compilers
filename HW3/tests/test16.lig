
// Creates a new array of n elements and initializes it to zero.
function new_array(n : int) : [int] {
      var arr : [int] = nil;
      var i : int;
      arr = resize(arr, n);
      for (i = 0 to n) {
            arr[i] = 0; 
      }
}

