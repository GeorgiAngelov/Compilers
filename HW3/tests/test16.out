(decls
   (decl-fun:ok (id new_array) (type-fun (params (typed-id (id n) (type-int))) (type-array (type-int)))
      (decls
         (decl-var:ok (id arr) (type-array (type-int)) (nil:nil))
         (decl-var:ok (id i) (type-int)))
      (stmts
         (assign:ok (id:[int] arr) (fun-call:[int] (id resize) (exps (id:[int] arr) (id:int n))))
         (for:ok (id:int i) (num:int 0) (id:int n)
            (stmts
               (assign:ok (array-idx:int (id:[int] arr) (id:int i)) (num:int 0)))))))
Function main() not defined or has wrong type!
